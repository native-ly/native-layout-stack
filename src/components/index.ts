export { Stack } from './Stack'
export { Spacer } from './Spacer'
