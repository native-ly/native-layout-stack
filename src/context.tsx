import React from 'react'

import type { Padding } from './types/Padding'

export interface Props {
  readonly debug: boolean
  readonly spaces?: number
  readonly padding?: Padding
}

export const LayoutContext = React.createContext<Props>({
  debug: false,
  spaces: undefined,
  padding: undefined,
})

LayoutContext.displayName = 'LayoutContext'

export const LayoutContextProvider: React.FC<Props> = ({
  children,
  ...props
}) => <LayoutContext.Provider value={props}>{children}</LayoutContext.Provider>
