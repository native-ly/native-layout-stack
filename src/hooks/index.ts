export { useStack } from './useStack'
export { useDebugStyle } from './useDebugStyle'
export { useSpacing } from './useSpacing'
export { useSpacingHelpers } from './useSpacingHelpers'
