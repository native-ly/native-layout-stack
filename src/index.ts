export { LayoutProvider } from './context'
export * from './components'
