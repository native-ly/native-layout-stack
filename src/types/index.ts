export type { Padding } from './Padding'
export type { Spaces } from './Spaces'
