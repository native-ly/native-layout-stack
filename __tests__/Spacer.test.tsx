import React from 'react'
import { Text } from 'react-native'
import { render } from '@testing-library/react-native'

import { Spacer, LayoutProvider } from '../src'

jest.mock('randomcolor', () => () => '#d259ff')

describe('Spacer', () => {
  it('should render component without context provider', () => {
    const { toJSON } = render(
      <>
        <Text>Hello</Text>

        <Spacer />

        <Text>World</Text>
      </>
    )

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render component with custom styles', () => {
    const { toJSON } = render(
      <>
        <Text>Hello</Text>

        <Spacer style={{ backgroundColor: 'orange' }} />

        <Text>World</Text>
      </>
    )

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render component with context provider', () => {
    const Component = () => (
      <>
        <Text>Hello</Text>

        <Spacer />

        <Text>World</Text>
      </>
    )

    const { toJSON } = render(
      <LayoutProvider>
        <Component />
      </LayoutProvider>
    )

    expect(toJSON()).toMatchSnapshot()
  })

  it('should render component with debug mode enabled', () => {
    const Component = () => (
      <>
        <Text>Hello</Text>

        <Spacer />

        <Text>World</Text>
      </>
    )

    const { toJSON } = render(
      <LayoutProvider debug>
        <Component />
      </LayoutProvider>
    )

    expect(toJSON()).toMatchSnapshot()
  })
})
