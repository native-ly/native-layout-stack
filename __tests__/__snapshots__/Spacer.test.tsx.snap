// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer should render component with context provider 1`] = `
Array [
  <Text>
    Hello
  </Text>,
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />,
  <Text>
    World
  </Text>,
]
`;

exports[`Spacer should render component with custom styles 1`] = `
Array [
  <Text>
    Hello
  </Text>,
  <View
    style={
      Object {
        "backgroundColor": "orange",
        "flex": 1,
      }
    }
  />,
  <Text>
    World
  </Text>,
]
`;

exports[`Spacer should render component with debug mode enabled 1`] = `
Array [
  <Text>
    Hello
  </Text>,
  <View
    style={
      Object {
        "backgroundColor": "#d259ff",
        "flex": 1,
      }
    }
  />,
  <Text>
    World
  </Text>,
]
`;

exports[`Spacer should render component without context provider 1`] = `
Array [
  <Text>
    Hello
  </Text>,
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  />,
  <Text>
    World
  </Text>,
]
`;
